cmd/crank/completion/completion.go

```diff
@@ -0,0 +1,259 @@
+		_, kubeconfig, err := kubernetesClient(parseConfigOverride(a))
+		if err != nil {
+			return nil
+		}
+
+		discoveryClient, err := discovery.NewDiscoveryClientForConfig(kubeconfig)
+		if err != nil {
+			return nil
+		}
+
+		resources, err := discoveryClient.ServerPreferredResources()
+		if err != nil {
+			return nil
+		}
+
+		if len(resources) == 0 {
+			return nil
+		}
+
+		var predictions []string
+		for _, apiResources := range resources {
+			for _, res := range apiResources.APIResources {
+				var resourceName string
+
+				switch {
+				case res.Group != "" && res.Version != "":
+					resourceName = fmt.Sprintf("%s.%s.%s", res.Name, res.Version, res.Group)
+				case res.Version != "":
+					resourceName = fmt.Sprintf("%s.%s", res.Name, res.Version)
+				default:
+					resourceName = res.Name
+				}
+
+				if strings.HasPrefix(resourceName, a.Last) {
+					predictions = append(predictions, resourceName)
+				}
+			}
+		}
+		return predictions
+	}
+}
+
+func kubernetesResourceNamePredictor() complete.PredictFunc {
+	return func(a complete.Args) []string {
+		client, kubeconfig, err := kubernetesClient(parseConfigOverride(a))
+		if err != nil {
+			return nil
+		}
+
+		discoveryClient, err := discovery.NewDiscoveryClientForConfig(kubeconfig)
+		if err != nil {
+			return nil
+		}
+
+		d := memory.NewMemCacheClient(discoveryClient)
+		rmapper := restmapper.NewShortcutExpander(restmapper.NewDeferredDiscoveryRESTMapper(d), d, nil)
+		mapping, err := mappingFor(rmapper, a.LastCompleted)
+		if err != nil {
+			return nil
+		}
+
+		// Get Resource object. Contains k8s resource and all its children, also as Resource.
+		u := &unstructured.UnstructuredList{}
+		u.SetGroupVersionKind(schema.GroupVersionKind{
+			Kind:    mapping.GroupVersionKind.Kind,
+			Group:   mapping.GroupVersionKind.Group,
+			Version: mapping.GroupVersionKind.Version,
+		})
+		err = client.List(context.Background(), u)
+		if err != nil {
+			return nil
+		}
+
+		var predictions []string
+		for _, res := range u.Items {
+			if strings.HasPrefix(res.GetName(), a.Last) {
+				predictions = append(predictions, res.GetName())
+			}
+		}
+		return predictions
+	}
+}
+
+func contextPredictor() complete.PredictFunc {
+	return func(a complete.Args) []string {
+		clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			clientcmd.NewDefaultClientConfigLoadingRules(),
+			&clientcmd.ConfigOverrides{},
+		)
+
+		kubeConfig, err := clientConfig.RawConfig()
+		if err != nil {
+			return nil
+		}
+
+		var predictions []string
+		for name := range kubeConfig.Contexts {
+			if strings.HasPrefix(name, a.Last) {
+				predictions = append(predictions, name)
+			}
+		}
+		return predictions
+	}
+}
+
+func namespacePredictor() complete.PredictFunc {
+	return func(a complete.Args) []string {
+		client, err := kubernetesClientset()
+		if err != nil {
+			return nil
+		}
+
+		namespaceList, err := client.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
```

Comment:
we use `context.Background()` earlier in this file, want to make that consistent here too?